File: articles/perma-model-of-well-being/fortran/perma_indices.f90
program perma_indices
  implicit none
  real :: positive_emotion, engagement, relationships, meaning, accomplishment
  real :: flourishing_score, life_satisfaction, institutional_support, institutional_barriers
  real :: autonomy_support, fairness_score, psychological_safety, access_score, workload_strain
  real :: perma, institutional_quality, context_adjusted

  positive_emotion = 6.9
  engagement = 7.1
  relationships = 7.2
  meaning = 7.0
  accomplishment = 7.1
  flourishing_score = 7.1
  life_satisfaction = 7.0
  institutional_support = 7.2
  institutional_barriers = 3.1
  autonomy_support = 7.1
  fairness_score = 7.0
  psychological_safety = 7.2
  access_score = 7.1
  workload_strain = 3.4

  perma = (positive_emotion + engagement + relationships + meaning + accomplishment) / 5.0
  institutional_quality = institutional_support + autonomy_support + fairness_score + &
      psychological_safety + access_score - institutional_barriers - workload_strain
  context_adjusted = flourishing_score + life_satisfaction + perma + institutional_quality

  print *, "PERMA index:", perma
  print *, "Institutional quality:", institutional_quality
  print *, "Context-adjusted flourishing:", context_adjusted
end program perma_indices

File: articles/perma-model-of-well-being/go/perma_indices.go
package main

import "fmt"

type PermaIndicators struct {
	PositiveEmotion      float64
	Engagement           float64
	Relationships        float64
	Meaning              float64
	Accomplishment       float64
	FlourishingScore     float64
	LifeSatisfaction     float64
	InstitutionalSupport float64
	InstitutionalBarrier float64
	AutonomySupport      float64
	FairnessScore        float64
	PsychologicalSafety  float64
	AccessScore          float64
	WorkloadStrain       float64
}

func PermaIndex(x PermaIndicators) float64 {
	return (x.PositiveEmotion + x.Engagement + x.Relationships + x.Meaning + x.Accomplishment) / 5.0
}

func InstitutionalQuality(x PermaIndicators) float64 {
	return x.InstitutionalSupport + x.AutonomySupport + x.FairnessScore +
		x.PsychologicalSafety + x.AccessScore -
		x.InstitutionalBarrier - x.WorkloadStrain
}

func ContextAdjustedFlourishing(x PermaIndicators) float64 {
	return x.FlourishingScore + x.LifeSatisfaction + PermaIndex(x) + InstitutionalQuality(x)
}

func main() {
	example := PermaIndicators{6.9, 7.1, 7.2, 7.0, 7.1, 7.1, 7.0, 7.2, 3.1, 7.1, 7.0, 7.2, 7.1, 3.4}

	fmt.Printf("PERMA index: %.3f\n", PermaIndex(example))
	fmt.Printf("Institutional quality: %.3f\n", InstitutionalQuality(example))
	fmt.Printf("Context-adjusted flourishing: %.3f\n", ContextAdjustedFlourishing(example))
}
